### PR TITLE
use envoy version with segfault fix

### PR DESCRIPTION
**Pull request by MarcinFalkowski**
Use envoy with a fix https://github.com/envoyproxy/envoy/pull/14092 backported to envoy v1.16.0

### CODE REVIEW

**Review by lukidzi — APPROVED** [review]

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoy/EnvoyContainer.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ class EnvoyContainer(
 
         const val EGRESS_LISTENER_CONTAINER_PORT = 5000
         const val INGRESS_LISTENER_CONTAINER_PORT = 5001
-        const val DEFAULT_IMAGE = "envoyproxy/envoy-alpine:v1.14.3"
+        const val DEFAULT_IMAGE = "marcinfalkowski/envoy-dev:v1.16.1-dev-lua-segfault-fix-1-16-0-backport-20201118-df9dc819"
```
Comment (chemicL) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoy/EnvoyContainer.kt:35]: Can we fork it on official allegro account not to use your personal?

File: tools/envoy/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,11 @@
-FROM envoyproxy/envoy-alpine:v1.14.3
+FROM marcinfalkowski/envoy-dev:v1.16.1-dev-lua-segfault-fix-1-16-0-backport-20201118-df9dc819
 
 ENV PORT=9999:9999
 ENV PORT=80:80
 ENV ENVOY_NODE_ID=front-proxy-id
 ENV ENVOY_NODE_CLUSTER=front-proxy
-ENV ENVOY_EGRESS_LISTENER_PORT=80
-ENV ENVOY_INGRESS_LISTENER_PORT=8080
+ENV ENVOY_EGRESS_LISTENER_PORT=31000
```
Comment (slonka) [inline tools/envoy/Dockerfile:7]: Is the port change relevant / needed?

Comment (MarcinFalkowski) [reply tools/envoy/Dockerfile:7]: yes, new envoy container versions launch envoy from user `envoy`, not `root`, which has no access to low ports

**Review by lukidzi — APPROVED** [review]

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoy/EnvoyContainer.kt (hunk 1 above)
Comment (MarcinFalkowski) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoy/EnvoyContainer.kt:35]: I will do it in next PR, when our official fork will be ready

**Review by andrzejwaw — APPROVED** [review]

**Pull request merged**